### PR TITLE
Added in support for TxEn pin required when using RS485 interface

### PR DESCRIPTION
**Pull request by Eskibrew**
ESP32 - Added in support for TxEn pin required when using RS485 interface.

---
This change is [Reviewable](https://reviewable.io/reviews/cesanta/mongoose-os/256)

### CODE REVIEW

File: fw/src/mgos_uart.h [hunk 1]
```diff
@@ -150,10 +153,16 @@ void mgos_uart_config_set_rx_params(struct mgos_uart_config *cfg,
 
 /*
  * Set Tx params in the provided config structure: buffer size `tx_buf_size`
- * and whether Tx flow control is enabled (`tx_fc_ena`).
+ * and whether Tx flow control is enabled (`tx_fc_ena`)
+ * 
+ * and whether TxEn (TxEnable) pin is enabled for RS485 mode (`rs485_ena`) where
+ * date is transmitted and received via the same twisted pair (half duplex).
```
Comment (nebulous) [inline fw/src/mgos_uart.h]: `data` is transmitted perhaps?

Comment (Eskibrew) [inline fw/src/mgos_uart.h]: well spotted - yes data.
I've actually made a further improvement so it's probably worth me issuing another PR before you accept this.

**Comment by cpq** [comment]
@Eskibrew would you be willing to create a dedicated library for RS485, as per https://mongoose-os.com/docs/core_components/apps.html#how-to-create-a-new-library ?

If done so, all Mongoose OS users can use it.

**Comment by Eskibrew** [comment]
I'd say yes however it's not obvious how it could be abstracted in that way since it needs modification of the platform specific files.  Am I missing something?

**Comment by cpq** [comment]
Correct, 485 functionality needs to be in the core

**Comment by tusharvb19** [comment]
Can you give example code to fetch data from modbus server i.e to act as a master?
TIA.

**Comment by tusharvb19** [comment]
Help needed...how can I poll modbus registers and coil i.e. act as master and use the rs485 ttl hardware with esp32 and 8266???
Or any arduino code ported?
TIA.

**Comment by Eskibrew** [comment]
This modification is not because I need MODBUS, I just need RS485, they are not synonymous.  I have in the past used [this library](https://github.com/webconn/freemodbus/blob/master/README.md) to achieve MODBUS master and slave operations on both PIC and MSP430 successfully.

**Comment by rojer** [comment]
i finally got around to it, and https://github.com/cesanta/mongoose-os/commit/a71bfbade5964601b004bc3f1de8d75616e46137 implements essentially this.

**Comment by yanzixiang** [comment]
maybe u can use this.
Half-Duplex RS-485/RS-422-Compatible Transceiver with AutoDirection Control
https://www.maximintegrated.com/en/products/interface/transceivers/MAX13487E.html

**Pull request closed without merging**